core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -261,6 +263,11 @@ public static RecordState forId(byte id) {
      */
     private long endOffset;
 
+    /**
+     * We maintain the latest fetch offset metadata to estimate the minBytes requirement more efficiently.
+     */
+    private Optional<LogOffsetMetadata> latestFetchOffsetMetadata;
```

Comment:
Why it's optional?

Reply:
I did it because the value can be null, so we thought it would be better to keep it as Optional https://github.com/apache/kafka/pull/17539#discussion_r1815520277